lib/new_relic/agent/instrumentation/action_mailbox_subscriber.rb

```diff
@@ -0,0 +1,66 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'new_relic/agent/instrumentation/notifications_subscriber'
+
+module NewRelic
+  module Agent
+    module Instrumentation
+      class ActionMailboxSubscriber < NotificationsSubscriber
+        def start(name, id, payload)
+          return unless state.is_execution_traced?
+
+          start_segment(name, id, payload)
+        rescue => e
+          log_notification_error(e, name, 'start')
+        end
+
+        def finish(name, id, payload)
+          return unless state.is_execution_traced?
+
+          finish_segment(id, payload)
+        rescue => e
+          log_notification_error(e, name, 'finish')
+        end
+
+        def start_segment(name, id, payload)
+          segment = Tracer.start_segment(name: metric_name(name, payload))
+          segment.params[:key] = payload[:key]
+          segment.params[:exist] = payload[:exist] if payload.key?(:exist)
+          push_segment(id, segment)
+        end
+
+        def finish_segment(id, payload)
+          if segment = pop_segment(id)
+            if exception = exception_object(payload)
+              segment.notice_error(exception)
+            end
+            segment.finish
+          end
+        end
+
+        def metric_name(name, payload)
+          mailbox = payload[:mailbox].class.name
+          method = method_from_name(name)
+          "Ruby/ActionMailbox/#{mailbox}/#{method}"
+        end
+
+        PATTERN = /\A([^\.]*)\.action_mailbox\z/
+        UNKNOWN = "unknown".freeze
+
+        METHOD_NAME_MAPPING = Hash.new do |h, k|
+          if PATTERN =~ k
+            h[k] = $1
```

Comment:
What is the `$1` doing?

Reply:
The `$1` here is referring to the first regex capture.

```ruby
'hannah' =~ /(han|nah)/
puts $1
  # => 'han'
```
To prevent confusion, Rubocop recommends using `Regexp.last_match(1)` instead of `$1`. Given that `$1` is already used by the other subscriber classes, I'm keeping it for now until we take a pass at DRYing and refactoring all subscriber classes post milestone.